common/include/pcl/impl/point_types.hpp

```diff
@@ -2417,6 +2454,33 @@ namespace traits
   template <typename PointT>
   using HasNoColor = std::enable_if_t<!has_color_v<PointT>, bool>;
 
+  /** Metafunction to check if a given point type has either rgb or rgba field. */
+  template <typename PointT>
+  struct has_rgb : has_any_field<PointT, boost::mpl::vector<pcl::fields::rgb,
+                                                              pcl::fields::rgba> >
+  { };
+
+  template <typename PointT>
+  constexpr auto has_rgb_v = has_rgb<PointT>::value;
+
+  template <typename PointT>
+  using HasRGB = std::enable_if_t<has_rgb_v<PointT>, bool>;
+
+  template <typename PointT>
+  using HasNoRGB = std::enable_if_t<!has_rgb_v<PointT>, bool>;
+
+  /** Metafunction to check if a given point type has any color (e.g. rgb, rgba, hsv, lab) field. */
+  template <typename PointT>
+  constexpr auto has_any_color_v = has_rgb_v<PointT> || has_HSV_v<PointT> || has_Lab_v<PointT> || has_intensity_v<PointT>;
+
+  template <typename PointT>
+  using HasAnyColor = std::enable_if_t<has_any_color_v<PointT>, bool>;
+
+  // There should be a `HasNoAnyColor` implementation which exactly means no color fields in point types,
+  // but the name seems to be obscure, so we decide to change the behavior of `HasNoColor` for this purpose
+  // as soon as the deprecation of `has_color` families in PCL 1.15.
+  // See here: https://github.com/PointCloudLibrary/pcl/pull/4761#discussion_r646427755
```

Comment:
Code is better than words. We can shift `HasNoColor` from L2418 (old) here, and stick the same deprecation notice on it: `[[deprecated("Please use HasNoRGB instead. Behavior will change in PCL 1.15")]]` or similar

Reply:
Thanks for your reviews! :grinning: 

Shifting `HasNoColor` is a good idea. But I think letting `HasNoColor`stay with `has_color` family (which are deprecated together) would give more convenience for further code changes. Thus, I shift all the `has_color` family after the `has_any_color` parts(i.e., after those comments).